### DISCUSSION TITLE
Is programmatic button image switching feasible for next-page, prev-page and profile-change button?

### DISCUSSION THREAD

**Discussion by kumazo007** [elgatosf/elgato#142] [Help]
Is programmatic button image switching feasible for next-page, prev-page and profile-change button?
As you know, those buttons are provided by Elgato.
Although I was looking for a way to do that, I haven't found that so far. Unfortunately,
My idea is to indicate to the operator to change SD's page from my plugin.
I would appreciate it if someone gave me hints.

Thanks,

**Comment by ChekTek** [comment]
Hi @kumazo007,

There is no way to fetch the current page or profile from Stream Deck at this time. Your plugin's action can listen for the [willAppear](https://docs.elgato.com/streamdeck/sdk/guides/actions/#onwillappear) event which fires when the key enters the canvas (i.e. when the page/profile it is on is visible) and the [willDisappear](https://docs.elgato.com/streamdeck/sdk/guides/actions/#onwilldisappear) event, which fires when the key leaves the canvas. You can then use [setImage](https://docs.elgato.com/streamdeck/sdk/guides/keys#setimage) to update the image.

You can switch profiles, and pages within the profile, if you're plugin owns the profile, however it is not possible for user created profiles. 

https://docs.elgato.com/streamdeck/sdk/guides/profiles

**Reply by kumazo007** [reply]
Hi @ChekTek 

I appreciated your kindly comment.
I developed my own plugin with the old SDK.  I probably need to use the New SDK that must be used for switching pages or profiles that you described. I hope Elgato will open those APIs for developers in the future.
Thanks a lot.

**Reply by ChekTek** [reply]
While I do recommend updating to the new SDK, you can absolutely still access the same API from the plugin using the old SDK, it would take some additional work to reference the websocket connection. The underlying websocket API is documented [here](https://docs.elgato.com/streamdeck/sdk/references/websocket/plugin), if you would like to explore!

**Reply by kumazo007** [reply]
@ChekTek 

It's good news to hear from you that I can use that feature in old SDK. I will try to do that.
Thanks a lot again.
